src/libraries/Common/src/System/Net/Security/SslClientAuthenticationOptionsExtensions.cs

```diff
@@ -33,7 +33,9 @@ public static SslClientAuthenticationOptions ShallowClone(this SslClientAuthenti
 
 #if DEBUG
             // Try to detect if a property gets added that we're not copying correctly.
-            foreach (PropertyInfo pi in typeof(SslClientAuthenticationOptions).GetProperties(BindingFlags.Public | BindingFlags.Instance | BindingFlags.DeclaredOnly))
+            PropertyInfo[] properties = typeof(SslClientAuthenticationOptions).GetProperties(BindingFlags.Public | BindingFlags.Instance | BindingFlags.DeclaredOnly)!;
+            Debug.Assert(properties.Length == 13);
```

Comment:
This was missed few times so I expect this would never hit in CI. Since there is dedicated test now, we can maybe move all that logic there...??? I only did minimal change discussed on https://github.com/dotnet/runtime/pull/88155 but we can possibly do much bigger change.

Reply:
Why did the existing debug validation below not catch the addition of AllowTlsResume?  Were we missing a test for AlowsTlsResume with HttpClient?

Reply:
because the clone would work fine for default values. And we did not have explicit test in Security.